### PR TITLE
Add an event when the imodelconnection changes.

### PR DESCRIPTION
**Pull request by a-gagnon**
## Changes
Since the usage of the store and `SessionStateActionId.SetIModelConnection` has been deprecated, I'm looking for an alternative to listen to IModel change event. Note that I can't use the `useActiveIModelConnection` hook as some of our code lives outside the React world, yet we still want to make sure we're targeting the same imodel as the ui framework.

Ideally, this is something that should be backported to the 4.16.x branch.

## Testing

Added a few lines in UiFramework.test.ts to validate that the event does indeed get called.

### CODE REVIEW

**Review by GerardasB — APPROVED** [review]
Please update `NextVersion.md`. You'll need to run `rush change` as well.

**Comment by a-gagnon** [comment]
> Please update `NextVersion.md`. You'll need to run `rush change` as well.

Done. I had a bunch of issues due to having a different pnpmVersion installed locally. Had to do my initial commit using the GH file edit.

**Comment by GerardasB** [comment]
> > Please update `NextVersion.md`. You'll need to run `rush change` as well.
> 
> Done. I had a bunch of issues due to having a different pnpmVersion installed locally. Had to do my initial commit using the GH file edit.

Sorry about that, it must have been due to https://github.com/iTwin/appui/pull/1025

File: common/changes/@itwin/appui-react/task-add-imodel-changed-event_2024-09-16-16-36.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@itwin/appui-react",
+      "comment": "Add an event (onImodelConnectionChanged) to get notified whenever the target imodelconnection changes.",
```
Comment (GerardasB) [inline common/changes/@itwin/appui-react/task-add-imodel-changed-event_2024-09-16-16-36.json:5]:
```suggestion
      "comment": "Added `UiFramework.onIModelConnectionChanged` event to get notified whenever the iModel connection returned by `UiFramework.getIModelConnection()` is changed.",
```

File: docs/changehistory/NextVersion.md [hunk 2]
```diff
@@ -14,6 +14,10 @@ Table of contents:
 
 - Allow to set the available snap modes in `SnapModeField` component. [#974](https://github.com/iTwin/appui/pull/974)
 
+### Additions
+
+- Added `UiFramework.onIModelConnectionChanged` event to get notified whenever the imodel targeted by `getIModelConnection` changes. This is a replacement for listening to the deprecated `SessionStateActionId.SetIModelConnection`.
```
Comment (GerardasB) [inline docs/changehistory/NextVersion.md:19]:
```suggestion
- Added `UiFramework.onIModelConnectionChanged` event to get notified whenever the iModel targeted by `UiFramework.getIModelConnection()` changes. This is a replacement for listening to the deprecated `SessionStateActionId.SetIModelConnection`.
```

**Review by GerardasB — APPROVED** [review]

**Pull request merged**